core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsInput.java

```diff
@@ -44,6 +54,78 @@ static GtfsInput createFromPath(String path) throws IOException {
         return new GtfsZipFileInput(file);
     }
 
+    /**
+     * Creates an specific GtfsInput to read data from the given URL.
+     *
+     * @param sourceUrl
+     * @param targetPathAsString
+     * @return
+     * @throws IOException
+     */
+    static GtfsInput createFromUrl(URL sourceUrl, String targetPathAsString) throws IOException {
+        InputStream inputStream;
+
+        try {
+            final HttpURLConnection httpConnection = (HttpURLConnection) sourceUrl.openConnection();
+            final int responseCode = httpConnection.getResponseCode();
+            // check response code
+            if (responseCode == HTTP_MOVED_PERM || responseCode == HTTP_MOVED_TEMP ||
+                    responseCode == HTTP_TEMP_REDIRECT || responseCode == HTTP_PERM_REDIRECT) {
+                // use redirection instead of original url
+                final String newUrlAsString = httpConnection.getHeaderField("Location");
+                final URLConnection connection = new URL(newUrlAsString).openConnection();
+                inputStream = connection.getInputStream();
+            } else {
+                inputStream = sourceUrl.openStream();
+            }
+            Path path = createPath(targetPathAsString);
+            Files.copy(
+                    inputStream,
+                    path,
+                    StandardCopyOption.REPLACE_EXISTING
+            );
+            return createFromPath(path.toString());
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Creates a path from a given string
+     * @param toCleanOrCreate
+     * @return
+     */
+    static Path createPath(final String toCleanOrCreate) {
+        // to empty any already existing directory
+        Path pathToCleanOrCreate = Paths.get(toCleanOrCreate);
+        if (Files.exists(pathToCleanOrCreate)) {
+            try {
+                //noinspection ResultOfMethodCallIgnored -- we ignore if deletion went well or not
+                Files.walk(pathToCleanOrCreate).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+                Files.createDirectory(pathToCleanOrCreate);
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Should we propagate all exceptions caught in this function? Is it really safe to silently ignore them all?

Reply:
Good catch. Most probably it isn't.

Reply:
Good point, it has been modified.

Reply:
@aababilov do you htink tht we should adopt a similar approach to what was done in `v1.4.0` which consists in: generating an notice that will be added to the `NoticeContainer`, after addition of said notice the validator would abort and export the validation report?

Reply:
This is a great question. I would suggest to generate a separate report for system errors (IO error, out of memory, exception raised in a validation etc). 

Rationale.
------------
1. Program that runs the validator needs to know why it crashed and a JSON report is more convenient than parsing stdout.
2. There may be bugs in individual validations and we need information about them.
3. We should not leak information about system errors to clients, so system errors should not be placed to the same report as validation errors.

Implementation
--------------------
We can have a special subclass of `Notice` (`SystemError`). NoticeContainer called on SystemError should keep it in a separate place.

Reply:
Opening an issue on this subject so that we can proceed to implementation in a separate PR.